### PR TITLE
5.3.x x509 rest backport

### PR DESCRIPTION
**Pull request by ruckc**
- this is an attempt at cherry-picking the PR#3457 commit 4fccc264826114e50ec48538ec322b0fd87f3e84 from master, and backporting onto 5.3.x while ensuring everything it touches still builds/checkstyles/findbugs cleanly.

### CODE REVIEW

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/core/rest/RestProperties.java [hunk 1]
```diff
@@ -31,4 +31,13 @@
      */
     private String attributeValue;
 
+    /**
+     * Flag that enables X509Certificate extraction from the request headers for authentication.
+     */
+    private boolean headerAuth = true;
```
Comment (mmoayyed) [inline api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/core/rest/RestProperties.java]: I believe this needs to be flipped to keep the same behavior as before.

**Comment by mmoayyed** [comment]
Looks excellent, thank you.

One small failure:

```
org.apereo.cas.support.x509.rest.X509RestHttpRequestHeaderCredentialFactoryTests > createX509Credential FAILED
    java.lang.AssertionError
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertFalse(Assert.java:64)
        at org.junit.Assert.assertFalse(Assert.java:74)
        at org.apereo.cas.support.x509.rest.X509RestHttpRequestHeaderCredentialFactoryTests.createX509Credential(X509RestHttpRequestHeaderCredentialFactoryTests.java:48)
```

...and minor comments here and there.

**Comment by ruckc** [comment]
I can't figure out why the test is failing, and for some reason gradle is skipping tests locally...

**Comment by mmoayyed** [comment]
OK, no worries. I'd wait for Travis CI to run through this batch once more if it still fails, we'll merge anyway and look into it later. Thanks!

**Comment by ruckc** [comment]
Yea, even updating the test and adding an @ToString shows the test "should" be succeeding.  its just a simple instanceof test like the other tests class.

**Comment by mmoayyed** [comment]
> org.apereo.cas.support.x509.rest.X509RestHttpRequestHeaderCredentialFactoryTests > createX509Credential FAILED
    java.lang.AssertionError: cred is not right type: X509CertificateCredential(certificates=[[

You may want to update the assert statement to indicate the credential type received; that way you can possibly try to figure out the reason for the mismatch.

**Comment by mmoayyed** [comment]
Also, shouldn't be assertTrue instead? You want to make sure the credential is in fact of that type, and else throw an error, right?

**Comment by ruckc** [comment]
your right, it should be assertTrue, on both test cases.

**Pull request merged**